wpcom-legacy-redirector.php

```diff
@@ -104,13 +105,58 @@ static function insert_legacy_redirect( $from_url, $redirect_to ) {
 			return new WP_Error( 'invalid-redirect-url', 'Invalid redirect_to param; should be a post_id or a URL' );
 		}
 
-		wp_insert_post( $args );
+		switch ( $mode ) {
+			case 'update':
+				$args[ 'ID' ] = self::get_redirect_post_id( $from_url, false );
```

Comment:
I think we can simplify most of this code by just using `wp_insert_post` and setting `$args['ID']`.